To: vim_dev@googlegroups.com
Subject: Patch 7.3.402
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.402
Problem:    When jumping to the first error a line of the buffer is sometimes
	    redrawn on top of the list of errors.
Solution:   Do not call update_topline_redraw() if the display was scrolled
	    up.
Files:	    src/quickfix.c


*** ../vim-7.3.401/src/quickfix.c	2012-01-10 17:13:48.000000000 +0100
--- src/quickfix.c	2012-01-20 13:13:25.000000000 +0100
***************
*** 1842,1849 ****
  #endif
  	if (print_message)
  	{
! 	    /* Update the screen before showing the message */
! 	    update_topline_redraw();
  	    sprintf((char *)IObuff, _("(%d of %d)%s%s: "), qf_index,
  		    qi->qf_lists[qi->qf_curlist].qf_count,
  		    qf_ptr->qf_cleared ? _(" (line deleted)") : "",
--- 1842,1851 ----
  #endif
  	if (print_message)
  	{
! 	    /* Update the screen before showing the message, unless the screen
! 	     * scrolled up. */
! 	    if (!msg_scrolled)
! 		update_topline_redraw();
  	    sprintf((char *)IObuff, _("(%d of %d)%s%s: "), qf_index,
  		    qi->qf_lists[qi->qf_curlist].qf_count,
  		    qf_ptr->qf_cleared ? _(" (line deleted)") : "",
*** ../vim-7.3.401/src/version.c	2012-01-10 22:31:26.000000000 +0100
--- src/version.c	2012-01-20 13:17:50.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     402,
  /**/

-- 
If all you have is a hammer, everything looks like a nail.
When your hammer is C++, everything begins to look like a thumb.
			-- Steve Hoflich, comp.lang.c++

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
